### PR TITLE
fix geometry tests

### PR DESCRIPTION
**Pull request by bitobs**
Fixes the geometry tests and styling problems.

## Description

We need to use a well stablished floating point type to not have unnecessary precision problems.
I have changed all functions to float, I don't think a template is neccessary.

I have also corrected the style of the { } in functions.

## How Has This Been Tested?

`dub test`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: source/geometry.d [hunk 1]
```diff
@@ -5,24 +5,27 @@ import std.math;
...
+pure Vector normalize(Vector v)
+{
+    Vector normalized =  v[] / magnitude (v);
+    return normalized;
+}
 
 unittest
 {
-    // Consider floating point accuracy problems.
+    // Consider floating point *inexistent* accuracy problems.
```
Comment (davidomarf) [inline source/geometry.d]: Being accuracy problems inexistent,
- L41 can test all events in magnitudes. Making L42-L45 useless.

File: source/geometry.d [hunk 2]
```diff
@@ -35,12 +38,18 @@ unittest
...
+    assert(magnitudes[1] == sqrt(14.0f));
+    assert(magnitudes[2] == sqrt(3.0f));
+    assert(magnitudes[3] == 0.0f);
    
     // Normalization
-    auto normalized = map!normalize(vectors).array;
-    // float[][] result = normalized [] - [cast(float) (3.0 / sqrt(14.0)), cast(float) (1.0 /sqrt(14.0)),cast(float) sqrt(2.0/7.0)];
+    auto normalizedVectors = map!normalize(vectors);
+
+    Vector dv0 = normalizedVectors[0][] - [3.0f/sqrt(14.0f), 1.0f/sqrt(14.0f), 2.0f/sqrt(14.0f)];
```
Comment (davidomarf) [inline source/geometry.d]: What's the advantage of using `dv0`, `dv1`, instead of comparing equality between vectors right away?

Comment (bitobs) [reply source/geometry.d]: Already done in local repository.

File: source/geometry.d [hunk 3]
```diff
@@ -35,12 +38,18 @@ unittest
 
     // Magnitude
     auto magnitudes = map!magnitude(vectors);
```
Comment (davidomarf) [inline source/geometry.d]: We should use one `expectedMagnitudes` array, and implement a utils function to compare content between two (will we ever use more than two to compare? could be useful?) arrays.

That way we would just run assert (arrayCompare(expected, received)).

Comment (davidomarf) [reply source/geometry.d]: Let's implement those.

File: source/geometry.d (hunk 2 above)
Comment (davidomarf) [reply source/geometry.d]: Commit it.

**Pull request merged**